src/git.rs

```diff
@@ -37,13 +48,39 @@ fn get_branch_info(
     })
 }
 
-pub fn git_list_branches(root: &Path) -> Result<Vec<BranchInfo>, git2::Error> {
+fn io_err_to_git_err(err: std::io::Error) -> git2::Error {
+    git2::Error::from_str(&err.to_string())
+}
+
+pub fn git_list_branches(root: &Path) -> Result<BranchResponse, git2::Error> {
     let repo = Repository::open(root)?;
-    let git_branches: Result<Vec<BranchInfo>, git2::Error> = repo
+
+    let default_branch_name = read_config(root)
+        .map_err(io_err_to_git_err)?
+        .git
+        .default_branch;
+
+    let default_branch_struct = match default_branch_name {
```

Comment:
if im understanding this correctly, do you mean set `default_branch` in the `read_config` function to a default value? i think because we are trying both `main` and `master` it seems weird to choose one